torch/onnx/symbolic_helper.py

```diff
@@ -712,6 +712,16 @@ def _index_fill_reshape_helper(g, self, dim, index):
     expanded_index = expand(g, unsqueezed_index, expanded_index_shape, None)
     return expanded_index_shape, expanded_index
 
+# When using reshape helper (opset_version >= 14), if reshape has -1,
+# allowzero cannot be set to 1
+def _reshape_helper(g, input, shape, allowzero=1):
```

Comment:
Seems it is more reasonable to set `allowzero=0` by default, because this is the default value for Reshape opset 14. Then we don't need have `allowzero=0` in any g.op("Reshape") call in old opsets.

Reply:
The reason allowzero=1 by default is because pytorch default behavior is reshape with allowzero=1